File: frontend/src/types/duplicate.ts
import type { TrackStatus } from './track'

export interface DuplicateGroupMember {
  track_id: number
  status: TrackStatus
  source_path: string
  final_path: string | null
  fingerprint_hash: string
  duration_seconds: number | null
  format_extension: string | null
  bitrate_kbps: number | null
  artist: string | null
  title: string | null
  integrated_lufs: number | null
}

export interface DuplicateGroup {
  id: number
  fingerprint_hash: string
  preferred_track_id: number | null
  members: DuplicateGroupMember[]
}

export interface DuplicateResolveRequest {
  group_id: number
  keep_track_id: number
  archive_track_ids?: number[]
}

export interface DuplicateResolveResponse {
  status: string
  kept_track_id: number | null
  archived_track_ids: number[]
}

File: backend/app/schemas/duplicate.py
from pydantic import BaseModel

from app.models.enums import TrackStatus


class DuplicateGroupMember(BaseModel):
    track_id: int
    status: TrackStatus
    source_path: str
    final_path: str | None
    fingerprint_hash: str
    duration_seconds: float | None
    format_extension: str | None
    bitrate_kbps: int | None


class DuplicateGroupResponse(BaseModel):
    id: int
    fingerprint_hash: str
    members: list[DuplicateGroupMember]


class DuplicateResolveResponse(BaseModel):
    status: str
